src/NumericalAlgorithms/Interpolation/InterpolationTargetKerrHorizon.hpp

```diff
@@ -0,0 +1,166 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <pup.h>
+
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "ApparentHorizons/Tags.hpp"
+#include "ApparentHorizons/YlmSpherepack.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrHorizon.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace db {
+template <typename TagsList>
+class DataBox;
+}  // namespace db
+namespace intrp {
+namespace Tags {
+template <typename Metavariables>
+struct TemporalIds;
+}  // namespace Tags
+}  // namespace intrp
+/// \endcond
+
+namespace intrp {
+
+namespace OptionHolders {
+/// A surface that conforms to the horizon of a Kerr black hole in
+/// Kerr-Schild coordinates.
+struct KerrHorizon {
+  struct Lmax {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "KerrHorizon is expanded in Ylms up to l=Lmax"};
+  };
+  struct Center {
+    using type = std::array<double, 3>;
+    static constexpr OptionString help = {"Center of black hole"};
+  };
+  struct Mass {
+    using type = double;
```

Comment:
should probably be bounded to be positive. (I don't know what a kerr solution does otherwise...)